Assets/Script/Song/SongSources.cs

```diff
@@ -322,6 +318,56 @@ private static void ReadSources()
                     }
                 }
             }
+
+            // Read individual PNGs (this is done after reading regular sources, so their icons can be replaced if needed)
+            if (!customSourcesRead) {
+                PathHelper.SafeEnumerateFiles(CustomSourcesFolder, "*.png", true, (path) =>
+                {
+                    // Filename will be used for all values (id, name, icon path).
+                    var icon = Path.GetFileNameWithoutExtension(path);
+                    var names = new Dictionary<string, string> {
+                        { "en-US", icon }
+                    };
+                    var parsed = new ParsedSource(icon, names, SourceType.Custom);
+                    if (_sources.TryAdd(icon, parsed))
+                    {
+                        if (icon == DEFAULT_KEY)
+                        {
+                            _default = parsed;
+                        }
+                    }
+                    else // Source already exists; override its icon
+                    {
+                        _sources[icon]._icon = icon;
+                    }
```

Comment:
I believe this can be simplified to just unconditionally inserting the icon, regardless of if it existed previously:

```suggestion
                    _sources[icon] = parsed;
                    if (icon == DEFAULT_KEY)
                    {
                        _default = parsed;
                    }
```

Reply:
Not really because, if the source already exists, I want to preserve its _metadata_ and replace only the icon itself.

For instance, if the user supplies a custom `gh.png`, surely they still want to keep the source name "Guitar Hero" intact. But if I do `_sources[icon] = parsed;`, the source name will be changed to `gh` (the filename) as well. `_sources[icon]._icon = icon;` will change the icon whilst preserving the metadata.

(This is also why I had to change `_icon` from `private` to `public` as well)

Reply:
Ah I didn't think that through all the way lol, all good here then.

I would have used `TryGetValue` to probe for existence instead of `TryAdd`, but the difference is minimal for a one-and-done operation, and so nitpicking the allocation efficiency of new `ParsedSource` instances isn't really necessary.